File: dashboards/python/routes/skills.py
"""
Skills routes: browse builtin skills and manage installed community skills.
"""

from flask import Blueprint, render_template, request, session, redirect, url_for, flash
from functools import wraps
from utils.api import api_request

skills_bp = Blueprint('skills_bp', __name__)


def login_required(f):
    @wraps(f)
    def decorated(*args, **kwargs):
        if 'token' not in session:
            return redirect(url_for('auth_bp.login'))
        return f(*args, **kwargs)
    return decorated


@skills_bp.route('/skills')
@login_required
def skills():
    categories_data = api_request('GET', '/engine/skills/by-category')
    installed_data = api_request('GET', '/engine/community/installed?orgId=default')
    return render_template(
        'skills.html',
        active_page='skills',
        user=session.get('user', {}),
        categories=categories_data.get('categories', {}),
        installed=installed_data.get('skills', []),
    )


@skills_bp.route('/skills/enable', methods=['POST'])
@login_required
def enable_skill():
    skill_id = request.form['skill_id']
    result = api_request('POST', f'/engine/community/{skill_id}/enable')
    flash('Skill enabled!' if not result.get('error') else result.get('error', 'Failed'))
    return redirect(url_for('skills_bp.skills'))


@skills_bp.route('/skills/disable', methods=['POST'])
@login_required
def disable_skill():
    skill_id = request.form['skill_id']
    result = api_request('POST', f'/engine/community/{skill_id}/disable')
    flash('Skill disabled!' if not result.get('error') else result.get('error', 'Failed'))
    return redirect(url_for('skills_bp.skills'))


@skills_bp.route('/skills/uninstall', methods=['POST'])
@login_required
def uninstall_skill():
    skill_id = request.form['skill_id']
    result = api_request('DELETE', f'/engine/community/{skill_id}')
    flash('Skill uninstalled!' if not result.get('error') else result.get('error', 'Failed'))
    return redirect(url_for('skills_bp.skills'))

File: dashboards/ruby/routes/skills.rb
# frozen_string_literal: true

# Skills routes — builtin skills browser + community skill management

get '/skills' do
  require_auth!
  categories_res = api_get('/engine/skills/by-category')
  @categories = categories_res[:status] == 200 ? (categories_res[:body]['categories'] || categories_res[:body]) : {}
  @categories = @categories.is_a?(Hash) ? @categories : {}
  installed_res = api_get('/engine/community/installed?orgId=default')
  @installed = installed_res[:status] == 200 ? (installed_res[:body]['skills'] || installed_res[:body]) : []
  @installed = @installed.is_a?(Array) ? @installed : []
  erb :skills
end

post '/skills/enable' do
  require_auth!
  skill_id = params[:skill_id]
  res = api_post("/engine/community/#{skill_id}/enable", {})
  if res[:status] < 300
    set_flash('Skill enabled successfully', 'success')
  else
    set_flash(res[:body].is_a?(Hash) ? (res[:body]['error'] || 'Failed to enable skill') : 'Failed to enable skill', 'danger')
  end
  redirect '/skills'
end

post '/skills/disable' do
  require_auth!
  skill_id = params[:skill_id]
  res = api_post("/engine/community/#{skill_id}/disable", {})
  if res[:status] < 300
    set_flash('Skill disabled successfully', 'success')
  else
    set_flash(res[:body].is_a?(Hash) ? (res[:body]['error'] || 'Failed to disable skill') : 'Failed to disable skill', 'danger')
  end
  redirect '/skills'
end

post '/skills/uninstall' do
  require_auth!
  skill_id = params[:skill_id]
  res = api_delete("/engine/community/#{skill_id}")
  if res[:status] < 300
    set_flash('Skill uninstalled successfully', 'success')
  else
    set_flash(res[:body].is_a?(Hash) ? (res[:body]['error'] || 'Failed to uninstall skill') : 'Failed to uninstall skill', 'danger')
  end
  redirect '/skills'
end
